spec/requests/classified_listings_spec.rb

```diff
@@ -272,5 +326,140 @@
         expect(org_listing.reload.bumped_at >= previous_bumped_at).to eq(true)
       end
     end
+
+    context "when the publish action is called" do
+      let(:params) { { classified_listing: { action: "publish" } } }
+
+      it "publishes a draft and charges user credits if first publish" do
+        cost = ClassifiedListing.cost_by_category(listing_draft.category)
+        create_list(:credit, cost, user: user)
+        expect do
+          put "/listings/#{listing_draft.id}", params: params
+        end.to change(user.credits.spent, :size).by(cost)
+        expect(listing_draft.reload.published).to eq(true)
+      end
+
+      it "publishes a draft and charges org credits before user credits if first publish" do
+        cost = ClassifiedListing.cost_by_category(org_listing_draft.category)
+        create_list(:credit, cost, organization: organization)
+        expect do
+          put "/listings/#{org_listing_draft.id}", params: params
+        end.to change(organization.credits.spent, :size).by(cost)
```

Comment:
I'm a bit concerned about having 2 expectations in the tests. Ideally, we should test these things separately: publishing and charging. I don't think it's worth fixing all occurrences in this file, it's just a note for the future.

Reply:
That's a valid point, but in most cases, I still prefer to have multiple expectations only if they are related to each other, like checking that several attributes are set correctly. When separating expectations we can be sure what exactly fails (if it fails).